### PR TITLE
exclusion filters for service names and arg1

### PR DESCRIPTION
**Pull request by ShanniLi**
r @kriskowal @blampe @anson627 

Test with node/examples/example1.js

tcap.js -i lo0 -i en0 -p 4040 -p 4041 -p 8065 -s ~server
listening on interface lo0 with filter ip proto \tcp and (port 4040 or port 4041 or port 8065 or portrange 4000-6000)
listening on interface en0 with filter ip proto \tcp and (port 4040 or port 4041 or port 8065 or portrange 4000-6000)
session=0 started src=127.0.0.1:60221 --> dst=127.0.0.1:4040 on lo0
session=0 127.0.0.1:60221 --> 127.0.0.1:4040 frame=1 type=0x01
INIT REQUEST id=0x0001 (1) version=2
headers
  host_port: 0.0.0.0:0
  process_name: node[5085]

session=0 127.0.0.1:60221 <-- 127.0.0.1:4040 frame=1 type=0x02
INIT RESPONSE id=0x0001 (1) version=2
headers
  host_port: 127.0.0.1:4040
  process_name: node[5085]

session=0 ended src=127.0.0.1:60221 --> dst=127.0.0.1:4040 on lo0

tcap.js -i lo0 -i en0 -p 4040 -p 4041 -p 8065 -p 4000-6000 -1 ~func1
listening on interface lo0 with filter ip proto \tcp and (port 4040 or port 4041 or port 8065 or portrange 4000-6000)
listening on interface en0 with filter ip proto \tcp and (port 4040 or port 4041 or port 8065 or portrange 4000-6000)
session=0 started src=127.0.0.1:61790 --> dst=127.0.0.1:4040 on lo0
session=0 127.0.0.1:61790 --> 127.0.0.1:4040 frame=1 type=0x01
INIT REQUEST id=0x0001 (1) version=2
headers
  host_port: 0.0.0.0:0
  process_name: node[6484]

session=0 127.0.0.1:61790 <-- 127.0.0.1:4040 frame=1 type=0x02
INIT RESPONSE id=0x0001 (1) version=2
headers
  host_port: 127.0.0.1:4040
  process_name: node[6484]

session=0 127.0.0.1:61790 --> 127.0.0.1:4040 frame=3 type=0x03
CALL REQUEST id=0x0003 (3) service="server" flags=0x00 ttl=0x0064 (100)
headers
  cn: example-client
  as: raw
  re: c
tracing: spanid=5627b7e9b22198f2 parentid=0000000000000000 traceid=5627b7e9b22198f2 flags=0x00
args[0]
    00: 6675 6e63 32                             func2
args[1]
    00: 6172 6720 31                             arg 1
args[2]
    00: 6172 6720 32                             arg 2

session=0 127.0.0.1:61790 <-- 127.0.0.1:4040 frame=3 type=0x04 NotOk
CALL RESPONSE id=0x0003 (3) flags=0x00
headers
  as: raw
tracing: spanid=5627b7e9b22198f2 parentid=0000000000000000 traceid=5627b7e9b22198f2 flags=0x00
args[0]
    00:                                          empty
args[1]
    00:                                          empty
args[2]
    00: 6974 2066 6169 6c65 64                   it failed

session=0 ended src=127.0.0.1:61790 --> dst=127.0.0.1:4040 on lo0

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,9 +20,11 @@ Uses pcap to inspect tchannel traffic over a network interface.
     -i --interface <interface>   network interface interfaces (defaults to first with an address)
     -p --port <port>             a port to track or use "port1-port2" for a range of ports to track between port1 and port2
     -f --filter <filter>         packet filter in pcap-filter(7) syntax (default: all TCP packets on port 4040)
-    -s --service <service-name>  service name or names to show (default: all services shown)
+    -s --service <service-name>  service name or names to show (default: all services shown), or
+                                 use "~service-name" to exclude the service
```
Comment (kriskowal) [inline README.md]: I think it would be okay and easier to use - instead of ~ here

File: frame-filters.js [hunk 2]
```diff
@@ -31,6 +32,24 @@
 var TChannelFrame = require('tchannel/v2/index');
 var FrameTypes = TChannelFrame.Types;
 
+
+function inclusionValidate(filters, filterName) {
+    var inclusion;
+    var exclusion;
```
Comment (kriskowal) [inline frame-filters.js]: Consider `inclusive` and `exclusive`.

**Comment by kriskowal** [comment]
great, :+1: and feedback optional.

File: README.md (hunk 1 above)
Comment (anson627) [inline README.md]: - looks like another new parameter, ~ is natural for negation

**Comment by ShanniLi** [comment]
all comments fixed.

File: README.md (hunk 1 above)
Comment (ShanniLi) [inline README.md]: yes, feel the same way, I will revert back to ~

**Comment by jcorbin** [comment]
Yeah, I've got to agree with the "~" camp, using "-" is too confusing with option markers

**Pull request merged**